python_modules/dagster/dagster/_core/pipes/context.py

```diff
@@ -475,6 +487,19 @@ def report_launched(self, launched_payload: PipesLaunchedData):
         """
         self.message_handler.on_launched(launched_payload)
 
+    def _inject_metadata_to_materialization_results(
+        self, results: Sequence[PipesExecutionResult], metadata: Mapping[str, Any]
+    ) -> Sequence[PipesExecutionResult]:
+        results_with_metadata = []
+        for i, result in enumerate(results):
+            if isinstance(result, MaterializeResult):
+                # .with_metadata returns a new instance
+                results_with_metadata.append(result.with_metadata(metadata))
+            else:
+                # do not modify asset checks
+                results_with_metadata.append(result)
```

Comment:
Why are we not modifying asset checks? Is it not useful to have e.g. the same `job_id` on them?

Reply:
Actually no idea why I decided to skip them. It makes sense to always inject this metadata. Probably brain lag